homeworks/veizerattila/5_oop/car_fleet_mgr.py

```diff
@@ -0,0 +1,116 @@
+﻿"""Hozz létre egy car_fleet_mgr.py nevű file-t, és kódold le a következő feladat megoldását:
+
+OK  - Készíts egy Car osztályt, amely rendelkezik a következő tulajdonságokkal:
+OK      ● Márka (brand)
+OK      ● Modell (model)
+OK      ● Gyártási év (year)
+OK      ● Kilométeróra állása (mileage), induló értéke 0.
+OK      ● Üzemanyag-szint (fuel_level), induló értéke 100 (százalékban).
+
+- Az osztály tartalmazza a következő metódusokat:
+OK      ● Egy konstruktor, amely beállítja a fenti attribútumokat.
+OK      ● Egy drive() metódus, amely adott számú kilométerrel növeli a kilométeróra állását, és csökkenti az üzemanyag-szintet (tételezzük fel, hogy 0.1% üzemanyag fogy megtett kilométerenként). A drive() metódus csak annyit km-et engedjen vezetni, amennyire elegendő üzemanyag van.
+OK      ● Egy refuel() metódus, amely feltölti az üzemanyag-szintet egy adott mennyiséggel. Figyelj a limitekre.
+
+OK  - Készíts egy Fleet osztályt, amely kezeli a Car objektumokat:
+OK      ● Az osztály rendelkezzen egy listával, amelyben az autók találhatóak.
+OK      ● Tartalmazzon metódusokat Car objektumok hozzáadására és eltávolítására a flottába/flottából.
+OK      ● Tartalmazzon egy metódust, amely összesíti a flotta összes autójának összes kilométerét.
+OK      ● Hozz létre néhány Car objektumot, add hozzá őket a flottához, hajts végre néhány műveletet (vezetés, tankolás), jelenítsd meg az autók állapotát és a flotta összesítő adatait."""
+
+##############################################################################
+######################## I. Osztályok létrehozása:############################
+##############################################################################
+
+
+######################## "Car" osztály létrehozása:###########################
+class Car:
+    # Konstruktor létrezhozása, ami beállítja a kért attribútomokat (márka, modell, év + km óra, benzin szint):
+    def __init__(self, brand: str, model: str, year: int):
+        self.brand = brand
+        self.model = model
+        self.year = year
+        self.mileage = 0                    # km
+        self.fuel_level = 100               # %
+
+    # Drive() metódus definiálása:
+    def drive(self, kilometers):
+        fuel_needed = kilometers * .1
+        if self.fuel_level > fuel_needed:
+            self.mileage += kilometers
+            self.fuel_level -= fuel_needed
+        else:
+            print(f'Nincs elég benzin.')
+
+    # Refuel() metódus definiálása:
+    def refuel(self, liter):
```

Comment:
Itt szuper lenne megvizsgálni hogy pozitív szám e a megadott liter, hogy ne tudják leszívni az üzemanyagot :)
Illetve alapvetően a mennyiség %-ot takar, nem litert, de ez igazából a megvalósításon nem változtat, ez nem hiba.

Reply:
Teljesen jogos, ezt is beépítettem egy külső if-else-be ágyazva.